safe/converter.py

```diff
@@ -110,8 +104,9 @@ def _find_branch_number(cls, inp: str):
         Args:
             inp: input smiles
         """
-
-        matching_groups = re.findall(r"((?<=%)\d{2})|((?<!%)\d+)", inp)
+        # Atom mapping case: avoid to capture brackets with :\d
+        inp = re.sub("\[[^:\]]*?\]", "", inp)  # noqa
```

Comment:
Just removing the brackets indeed cover our cases (first version in the issue is exactly like that), but I added the atom mapping case just in case to be "more precise" on the substitution. 
After testing I think is safe to avoid this case and just remove the brackets.